Tree.java

```diff
@@ -0,0 +1,107 @@
+/**Implemented tree data structure
+ * public data member: root
+ * public method: poplulateTree(int [] data) creates tree from data array  
+ */
+public class Tree{
+	static final int NullNode = Integer.MIN_VALUE;
+	Node<Integer> root = null;
+	
+	public Tree(){
+		this.root = null;
+	}
+	
+	/** populateTree(): Creates tree from data array
+	 * 
+	 * @param data
+	 * 
+	 * For each index i of the array, 
+	 * 		the data at index (2*i + 1) is the left child
+	 * 		the data at index (2*i + 2) is the right child
+	 */
+	public void populateTree(int data[]){
+		if(data == null || data.length == 0){
+			return;
+		}
+		this.root = new Node<Integer>(data[0]);
+		//System.out.println("Root: "+this.root.data);
+		createTree(this.root, 0, data);				
+	}
+	
+	private void createTree(Node<Integer> root, int n, int data[]){
+		int leftN = 2*n + 1;
+		int rightN = leftN + 1;
+		if(leftN >= data.length){
+			//System.out.println("Left of "+data[n]+" is not there and exceeds length");
+			root.left = null;

```

Comment:
I don't think you need to set the value of left and right, they will be null by default.